hh-dep-monitoring-service/src/main/java/ru/hh/school/depmonitoring/service/loaders/DependencyLoader.java

```diff
@@ -0,0 +1,130 @@
+package ru.hh.school.depmonitoring.service.loaders;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import ru.hh.school.depmonitoring.dao.ArtifactDao;
+import ru.hh.school.depmonitoring.dao.ArtifactVersionDao;
+import ru.hh.school.depmonitoring.dao.DependencyDao;
+import ru.hh.school.depmonitoring.dao.RepositoryDao;
+import ru.hh.school.depmonitoring.dto.bamboo.BambooArtifactDto;
+import ru.hh.school.depmonitoring.entities.Artifact;
+import ru.hh.school.depmonitoring.entities.ArtifactVersion;
+import ru.hh.school.depmonitoring.entities.Dependency;
+import ru.hh.school.depmonitoring.entities.Repository;
+
+import javax.inject.Named;
+import javax.inject.Singleton;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.core.GenericType;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Named
+@Singleton
+public class DependencyLoader {
+    private final String bambooLink;
+    private final RepositoryDao repositoryDao;
+    private final ArtifactDao artifactDao;
+    private final ArtifactVersionDao artifactVersionDao;
+    private final DependencyDao dependencyDao;
+
+    private static final Logger log = LoggerFactory.getLogger(DependencyLoader.class);
+
+    public DependencyLoader(
+            String bambooLink,
+            RepositoryDao repositoryDao,
+            ArtifactDao artifactDao,
+            ArtifactVersionDao artifactVersionDao,
+            DependencyDao dependencyDao
+    ) {
+        this.bambooLink = bambooLink;
+        this.repositoryDao = repositoryDao;
+        this.artifactDao = artifactDao;
+        this.artifactVersionDao = artifactVersionDao;
+        this.dependencyDao = dependencyDao;
+    }
+
+    private List<BambooArtifactDto> getDataFromBamboo() {
+        return ClientBuilder.newBuilder()
+                .build()
+                .target(bambooLink)
+                .request()
+                .get()
+                .readEntity(new GenericType<List<BambooArtifactDto>>() {
+                });
+    }
+
+    public void saveBambooDate() {
+        saveBambooDate(getDataFromBamboo(), null);
+    }
+
+    private void saveBambooDate(List<BambooArtifactDto> artifactDtoList, Repository parentRepository) {
+        for (BambooArtifactDto bambooArtifactDto: artifactDtoList) {
+            Repository repository = repositoryDao.findRepositoryByName(bambooArtifactDto.getRepoName()).orElse(null);
+
+            String[] keyList = bambooArtifactDto.getKey().split(":");
+            if (keyList.length != 3) {
+                log.error("Can't parse bamboo key to artifact name, group name and version (" + bambooArtifactDto.getKey() + ")");
```

Comment:
Логирование лучше не конкатенацией делать
```suggestion
                log.error("Can't parse bamboo key to artifact name, group name and version ({})", bambooArtifactDto.getKey());
```